### PR TITLE
GM: Auto-resume from stop-and-go

### PR DESCRIPTION
**Pull request by vntarasov**
Through resetting ACC (stopping sending ACC commands until it gets out of standstill) mode and immediately resuming. No need to press resume button anymore!

Piggybacking on @Jamezz's work on smoother braking from https://github.com/commaai/openpilot/pull/326.

### CODE REVIEW

**Comment by Jamezz** [comment]
Just verified this on my own branch. Works great!

**Comment by vntarasov** [comment]
auto-resume

File: selfdrive/car/gm/carcontroller.py [hunk 1]
```diff
@@ -133,12 +130,18 @@ def update(self, sendcan, enabled, CS, frame, actuators, \
       if (frame % 4) == 0:
         idx = (frame / 4) % 4
 
-        at_full_stop = enabled and CS.standstill
-        near_stop = enabled and (CS.v_ego < P.NEAR_STOP_BRAKE_PHASE)
+        car_stopping = apply_gas < P.ZERO_GAS
```
Comment (rbiasini) [inline selfdrive/car/gm/carcontroller.py]: what happens if you are stopping uphill and apply_gas remains positive?

Comment (vntarasov) [reply selfdrive/car/gm/carcontroller.py]: ah, didn't think of that (didn't test it on very steep hills)

**Comment by vntarasov** [comment]
Will reopen when steep hill logic (stopping with positive gas) is figured out.

File: selfdrive/car/gm/carcontroller.py (hunk 1 above)
Comment (Jamezz) [reply selfdrive/car/gm/carcontroller.py]: This is a fundamental issue with openpilot's interface design. The stopping intent is never passed up to an interface.

Comment (vntarasov) [reply selfdrive/car/gm/carcontroller.py]: Another problem is that GM interface doesn't know if car is moving forward or backwards.

**Pull request closed without merging**